design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -325,6 +325,30 @@ type CommentItem {
 }
 ```
 
+## OrderAddress type
+```graphql
+
+type OrderAddress @doc(description: "OrderAddress contains detailed information about the order billing and shipping addresses"){
```

Comment:
Are any of the fields in this type required in Magento? If any of these fields assigned to a scalar type are required in Magento, we should make them non-nullable here.

[More info](https://github.com/magento/architecture/blob/f18063cc1211399999a4c5caba0c7bdb2728cc27/design-documents/graph-ql/nullability.md#scalars-should-be-non-nullable-when-required-in-backing-data-source)

Reply:
<img width="1005" alt="AddressFields" src="https://user-images.githubusercontent.com/40216012/87596649-1c568080-c6b6-11ea-8d7a-e667ceb7d4b0.png">


<img width="1452" alt="AddressFields2" src="https://user-images.githubusercontent.com/40216012/87596898-82430800-c6b6-11ea-93c9-e208f30f2c2c.png">

The fields which we can make non-nullable mostly depend on the country we choose on the address. You can see in the above screenshots for more clear context. Probably, we can leave the fields 'Postcode', 'State', 'Street address' as nullable so that we make other fields as non-nullable. I am not sure about the 'City' field though. What do you think @DrewML?

Reply:
Thanks for the screenshots - super helpful.

> The fields which we can make non-nullable mostly depend on the country we choose on the address

For any cases where it seems ambiguous, we can play it safe and stick to nullable for now to avoid a breaking change later. Let's just do non-null for fields that are, in all cases, required.